plugins/node/opentelemetry-instrumentation-nestjs-core/src/utils.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// util.types.isPromise is supported from 10.0.0
+export const isPromise = (value: any): value is Promise<unknown> => {
```

Comment:
This seems like something that could be useful to more than one instrumentation. Possibly a good candidate for an instrumentation-utils package? Doesn't affect this PR

Reply:
Agreed. I'd more specifically 1. see as few of those utilities managed centrally as possible; 2. keep them in the instrumentation package itself.